internal/tools/orchestrator/scheduler/executor/plugins/k8s/deployment.go

```diff
@@ -1206,71 +1207,6 @@ func (k *Kubernetes) scaleDeployment(ctx context.Context, sg *apistructs.Service
 	return nil
 }
 
-func (k *Kubernetes) setContainerResources(service apistructs.Service, container *corev1.Container) error {
-	if service.Resources.Cpu < MIN_CPU_SIZE {
-		return errors.Errorf("invalid cpu, value: %v, (which is lower than min cpu(%v))",
-			service.Resources.Cpu, MIN_CPU_SIZE)
-	}
-
-	//Set the over-score ratio according to the environment
-	cpuSubscribeRatio := k.cpuSubscribeRatio
-	memSubscribeRatio := k.memSubscribeRatio
-	switch strutil.ToUpper(service.Env[types.DiceWorkSpace]) {
-	case "DEV":
-		cpuSubscribeRatio = k.devCpuSubscribeRatio
-		memSubscribeRatio = k.devMemSubscribeRatio
-	case "TEST":
-		cpuSubscribeRatio = k.testCpuSubscribeRatio
-		memSubscribeRatio = k.testMemSubscribeRatio
-	case "STAGING":
-		cpuSubscribeRatio = k.stagingCpuSubscribeRatio
-		memSubscribeRatio = k.stagingMemSubscribeRatio
-	}
-	if cpuSubscribeRatio < 1.0 {
-		cpuSubscribeRatio = 1.0
-	}
-	if memSubscribeRatio < 1.0 {
-		memSubscribeRatio = 1.0
-	}
-
-	cpu := fmt.Sprintf("%dm", int(service.Resources.Cpu*1000))
-	memory := fmt.Sprintf("%dMi", int(service.Resources.Mem))
-
-	maxCpu := fmt.Sprintf("%dm", int(service.Resources.MaxCPU*1000))
-	maxMem := fmt.Sprintf("%dMi", int(service.Resources.MaxMem))
-
-	container.Resources = corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceCPU:              resource.MustParse(cpu),
-			corev1.ResourceMemory:           resource.MustParse(memory),
-			corev1.ResourceEphemeralStorage: resource.MustParse(k8sapi.EphemeralStorageSizeRequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceCPU:              resource.MustParse(maxCpu),
-			corev1.ResourceMemory:           resource.MustParse(maxMem),
-			corev1.ResourceEphemeralStorage: resource.MustParse(k8sapi.EphemeralStorageSizeLimit),
-		},
-	}
-
-	if service.Resources.EphemeralStorageCapacity > 1 {
-		maxEphemeral := fmt.Sprintf("%dGi", service.Resources.EphemeralStorageCapacity)
-		container.Resources.Limits[corev1.ResourceEphemeralStorage] = resource.MustParse(maxEphemeral)
-	}
-
-	if err := k.SetFineGrainedCPU(container, map[string]string{}, cpuSubscribeRatio); err != nil {
-		logrus.Errorf("set cpu resource failed, container name: %s, error: %v", container.Name, err)
-		return err
-	}
-	if err := k.SetOverCommitMem(container, memSubscribeRatio); err != nil {
-		logrus.Errorf("set mem resource failed, container name: %s, error: %v", container.Name, err)
-		return err
-	}
-
-	logrus.Debugf("container name: %s, resource: %+v", container.Name, container.Resources)
-
-	return nil
-}
-
 func (k *Kubernetes) UpdateContainerResourceEnv(originResource apistructs.Resources, container *corev1.Container) {
 	for index, env := range container.Env {
 		var needToUpdate = false
```

Comment:
This file is too large, it should be split.

Reply:
Great idea! The code logic is now organized into separate functional modules.